File: sample-java/src/main/java/fr/xgouchet/mesmaeker/sample/DataWalker.java
package fr.xgouchet.mesmaeker.sample;

interface DataWalker<T> {

    void walkForward();

    void walkBackward();

    void walkTo(int position);

    T getCurrent();

    boolean hasNext();

    T peekNext();

    void setLoopEnabled(boolean enabled);

}

File: sample-kotlin/src/main/kotlin/fr/xgouchet/mesmaeker/sample/DataWalker.kt
package fr.xgouchet.mesmaeker.sample

interface DataWalker<T> {

    fun getCurrent(): T

    fun walkForward()

    fun walkBackward()

    fun walkTo(position: Int)

    fun hasNext(): Boolean

    fun peekNext(): T

    fun setLoopEnabled(enabled: Boolean)

}
